### PR TITLE
resource/aws_cloudwatch_metric_alarm: Add `period` to `metric_query` block

### LINKED ISSUE #28617: [Bug]: Cloudwatch Metric Alarm 'metric_query' block does not Support Period in Conjunction With Expression
### Terraform Core Version

1.3.6

### AWS Provider Version

4.48.0

### Affected Resource(s)

* aws_cloudwatch_metric_alarm

### Expected Behavior

When creating a Metric Alarm that uses an expression, it is not possible to set the period. However, this is a valid configuration in CloudWatch and can be successfuly performed via the web console and the API.

### Actual Behavior

This leads to the request being rejected by AWS as the 'period' field is missing in the request.

### Relevant Error/Panic Output Snippet

```shell
Error: Updating metric alarm failed: ValidationError: Period must not be null
│       status code: 400, request id: 95fc8690-15f5-433d-85f1-3a34e6516db8
│ 
│   with module.redshift_cluster.aws_cloudwatch_metric_alarm.wlm_total_average_queue_length[0],
│   on ../../../modules/redshift_cluster/main.tf line 378, in resource "aws_cloudwatch_metric_alarm" "wlm_total_average_queue_length":
│  378: resource "aws_cloudwatch_metric_alarm" "wlm_total_average_queue_length" {
```

### Terraform Configuration Files

```hcl
resource "aws_cloudwatch_metric_alarm" "wlm_total_average_queue_length" {
  alarm_name          = "redshift-archive-wlm-total-avg-queue-length"
  comparison_operator = "GreaterThanOrEqualToThreshold"
  evaluation_periods  = 1
  datapoints_to_alarm = 1
  threshold           = 5
  alarm_description   = "The WLM queue length across all queues has exceeded 5 on average in the last 15 minutes"
  
  metric_query {   
    id          = "avg_queue_length_all_queues"
    expression  = "SELECT AVG(WLMQueueLength) FROM \"AWS/Redshift\" WHERE ClusterIdentifier = 'archive'" # No QueueName dimension as we want to get the avg of all queue lengths
    label       = "Average queue length across all WLM queues"
    return_data = true
    // period      = 15 * 60 # Unable to set in TF
  }

  alarm_actions = ["arn:aws:sns:eu-west-1:582261275102:NotifyMe"]
}
```

### Steps to Reproduce

* Attempt to create a Cloudwatch Alarm with an expression

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

An example in Python showing the successful creation of such a metric:

```python
import boto3

def create_alarm(
    sns_topic_arn: str,
    redshift_cluster_identifier: str,
    evaluation_period_minutes: int=15,
    wlm_queue_total_average_max_acceptable_length: int=5
) -> None:
    cloudwatch = boto3.client('cloudwatch')

    alarm_data = {
        "AlarmName": f"redshift-{redshift_cluster_identifier}-wlm-total-avg-queue-length",
        "AlarmDescription": f"The WLM queue length across all queues has exceeded {wlm_queue_total_average_max_acceptable_length} on average in the last {evaluation_period_minutes} minutes",
        "ActionsEnabled": True,
        "AlarmActions": [sns_topic_arn],
        "EvaluationPeriods": 1,
        "DatapointsToAlarm": 1,
        "Threshold": wlm_queue_total_average_max_acceptable_length,
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "TreatMissingData": "missing",
        "Metrics": [
            {
                "Id": "avg_queue_length_all_queues",
                "Expression": f"SELECT AVG(WLMQueueLength) FROM \"AWS/Redshift\" WHERE ClusterIdentifier = '{redshift_cluster_identifier}'",
                "Label": "Average queue length across all WLM queues",
                "ReturnData": True,
                "Period": evaluation_period_minutes * 60
            }
        ]
    }

    cloudwatch.put_metric_alarm(**alarm_data)

REDSHIFT_CLUSTER_IDENTIFIER = "archive"
SNS_TOPIC_ARN = "arn:aws:sns:eu-west-1:582261275102:NotifyMe"

if __name__ == "__main__":
    create_alarm(sns_topic_arn=SNS_TOPIC_ARN, redshift_cluster_identifier=REDSHIFT_CLUSTER_IDENTIFIER)
```

### Would you like to implement a fix?

None

### LINKED ISSUE #29398: [Bug]: aws_cloudwatch_metric_alarm with expression throws: `"period": conflicts with metric_query`
### Terraform Core Version

1.3.1

### AWS Provider Version

4.32.0

### Affected Resource(s)

aws_cloudwatch_metric_alarm

### Expected Behavior

`terraform apply` with success 

### Actual Behavior

I get the error `"period": conflicts with metric_query` even though https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudwatch-alarm-metricdataquery.html mentions that period is now supported inside `MetricDataQuery`

Also using AWS Console I was able to create the alarm manually with the same expression as the one from above configuration.

The error seems to be due to `period` missing from https://github.com/hashicorp/terraform-provider-aws/blob/5857c145b31d41966bef256f669c12ae35cc2486/internal/service/cloudwatch/metric_alarm.go#L599

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```
resource "aws_cloudwatch_metric_alarm" "millis_behind" {
  alarm_name          = "MillisBehind alarm for ${var.app_name}"
  alarm_description   = "MillisBehind alarm for ${var.app_name}"
  comparison_operator = "GreaterThanOrEqualToThreshold"
  evaluation_periods  = "3"
  datapoints_to_alarm = "3"
  threshold           = 30000
  treat_missing_data  = "breaching"

  metric_query {
    id          = "mq-millis-behind"
    expression  = "SELECT MAX(MillisBehindLatest) FROM SCHEMA(\"${var.app_name}\", Operation, ShardId) WHERE Operation = 'ProcessTask'"
    period      = "60"
    return_data = "true"
  }
}
```

### Steps to Reproduce

Define `var.app_name` using the name of a KCL kubernates application and then create a new CW alarm with period inside meric_query (see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudwatch-alarm-metricdataquery.html for documentation).

### Debug Output

```
Error: Conflicting configuration arguments

  with module.metrics.aws_cloudwatch_metric_alarm.millis_behind,
  on ../../../../modules/kcl/cloudwatch.tf line 7, in resource "aws_cloudwatch_metric_alarm" "millis_behind":
   7:   period              = "60"

"period": conflicts with metric_query
```

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by gdavison**
### Description

Adds missing `period` parameter to `metric_query` block.

Also adds validation to `metric_query.metric.period`.

### Relations

Closes #29398.
Closes #28617.
Closes https://github.com/hashicorp/terraform-provider-aws/pull/28547.

### Output from Acceptance Testing

```
$ make testacc PKG=cloudwatch TESTS=TestAccCloudWatchMetricAlarm_

--- PASS: TestAccCloudWatchMetricAlarm_missingStatistic (8.67s)
--- PASS: TestAccCloudWatchMetricAlarm_disappears (17.45s)
--- PASS: TestAccCloudWatchMetricAlarm_dataPointsToAlarm (19.02s)
--- PASS: TestAccCloudWatchMetricAlarm_basic (23.23s)
--- PASS: TestAccCloudWatchMetricAlarm_AlarmActions_swfAction (23.46s)
--- PASS: TestAccCloudWatchMetricAlarm_AlarmActions_snsTopic (23.51s)
--- PASS: TestAccCloudWatchMetricAlarm_evaluateLowSampleCountPercentiles (30.60s)
--- PASS: TestAccCloudWatchMetricAlarm_treatMissingData (42.59s)
--- PASS: TestAccCloudWatchMetricAlarm_tags (42.77s)
--- PASS: TestAccCloudWatchMetricAlarm_metricQuery (143.82s)
--- PASS: TestAccCloudWatchMetricAlarm_AlarmActions_ec2Automate (177.82s)
--- PASS: TestAccCloudWatchMetricAlarm_extendedStatistic (270.39s)
```

### CODE REVIEW

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccCloudWatchMetricAlarm_' PKG=cloudwatch ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/cloudwatch/... -v -count 1 -parallel 3  -run=TestAccCloudWatchMetricAlarm_ -timeout 180m
=== RUN   TestAccCloudWatchMetricAlarm_basic
=== PAUSE TestAccCloudWatchMetricAlarm_basic
=== RUN   TestAccCloudWatchMetricAlarm_AlarmActions_ec2Automate
=== PAUSE TestAccCloudWatchMetricAlarm_AlarmActions_ec2Automate
=== RUN   TestAccCloudWatchMetricAlarm_AlarmActions_snsTopic
=== PAUSE TestAccCloudWatchMetricAlarm_AlarmActions_snsTopic
=== RUN   TestAccCloudWatchMetricAlarm_AlarmActions_swfAction
=== PAUSE TestAccCloudWatchMetricAlarm_AlarmActions_swfAction
=== RUN   TestAccCloudWatchMetricAlarm_dataPointsToAlarm
=== PAUSE TestAccCloudWatchMetricAlarm_dataPointsToAlarm
=== RUN   TestAccCloudWatchMetricAlarm_treatMissingData
=== PAUSE TestAccCloudWatchMetricAlarm_treatMissingData
=== RUN   TestAccCloudWatchMetricAlarm_evaluateLowSampleCountPercentiles
=== PAUSE TestAccCloudWatchMetricAlarm_evaluateLowSampleCountPercentiles
=== RUN   TestAccCloudWatchMetricAlarm_extendedStatistic
=== PAUSE TestAccCloudWatchMetricAlarm_extendedStatistic
=== RUN   TestAccCloudWatchMetricAlarm_metricQuery
=== PAUSE TestAccCloudWatchMetricAlarm_metricQuery
=== RUN   TestAccCloudWatchMetricAlarm_missingStatistic
=== PAUSE TestAccCloudWatchMetricAlarm_missingStatistic
=== RUN   TestAccCloudWatchMetricAlarm_tags
=== PAUSE TestAccCloudWatchMetricAlarm_tags
=== RUN   TestAccCloudWatchMetricAlarm_disappears
=== PAUSE TestAccCloudWatchMetricAlarm_disappears
=== CONT  TestAccCloudWatchMetricAlarm_basic
=== CONT  TestAccCloudWatchMetricAlarm_evaluateLowSampleCountPercentiles
=== CONT  TestAccCloudWatchMetricAlarm_missingStatistic
--- PASS: TestAccCloudWatchMetricAlarm_missingStatistic (6.75s)
=== CONT  TestAccCloudWatchMetricAlarm_extendedStatistic
--- PASS: TestAccCloudWatchMetricAlarm_basic (19.75s)
=== CONT  TestAccCloudWatchMetricAlarm_AlarmActions_swfAction
--- PASS: TestAccCloudWatchMetricAlarm_evaluateLowSampleCountPercentiles (29.34s)
=== CONT  TestAccCloudWatchMetricAlarm_treatMissingData
--- PASS: TestAccCloudWatchMetricAlarm_AlarmActions_swfAction (18.73s)
=== CONT  TestAccCloudWatchMetricAlarm_dataPointsToAlarm
--- PASS: TestAccCloudWatchMetricAlarm_dataPointsToAlarm (14.79s)
=== CONT  TestAccCloudWatchMetricAlarm_disappears
--- PASS: TestAccCloudWatchMetricAlarm_disappears (14.19s)
=== CONT  TestAccCloudWatchMetricAlarm_metricQuery
--- PASS: TestAccCloudWatchMetricAlarm_treatMissingData (44.17s)
=== CONT  TestAccCloudWatchMetricAlarm_AlarmActions_snsTopic
--- PASS: TestAccCloudWatchMetricAlarm_AlarmActions_snsTopic (18.98s)
=== CONT  TestAccCloudWatchMetricAlarm_AlarmActions_ec2Automate
--- PASS: TestAccCloudWatchMetricAlarm_metricQuery (92.89s)
=== CONT  TestAccCloudWatchMetricAlarm_tags
--- PASS: TestAccCloudWatchMetricAlarm_extendedStatistic (181.75s)
--- PASS: TestAccCloudWatchMetricAlarm_tags (38.07s)
--- PASS: TestAccCloudWatchMetricAlarm_AlarmActions_ec2Automate (230.41s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/cloudwatch	328.163s
```

**Comment by mvrogowski** [comment]
Hey, there! Does it have some estimation on this being released? Thanks!

**Comment by khanhnd0408** [comment]
I also met troubles with this; my workaround is to create on the AWS console and import it to Terraform state, but I think it may raise this error again when I want to create or change other resources.

File: internal/service/cloudwatch/metric_alarm.go [hunk 1]
```diff
@@ -113,6 +113,10 @@ func ResourceMetricAlarm() *schema.Resource {
 									"period": {
 										Type:     schema.TypeInt,
 										Required: true,
+										ValidateFunc: validation.Any(
+											validation.IntInSlice([]int{1, 5, 10, 30}),
```
[the remark this answers is not in the document]
Comment (gdavison) [reply internal/service/cloudwatch/metric_alarm.go:117]: Hi @henkka,the following line

```go
validation.IntDivisibleBy(60)
```

covers the multiples of 60 case. It's good to have extra eyes review things, though 🙂

[the remark this answers is not in the document]
Comment (henkka) [reply internal/service/cloudwatch/metric_alarm.go:117]: Ah right, my point was that this should actually be `validation.IntInSlice([]int{10, 30})` if we'd like to comply with what AWS allows currently

[the remark this answers is not in the document]
Comment (gdavison) [reply internal/service/cloudwatch/metric_alarm.go:117]: Ah, I see what you're saying. The top-level `period`, at https://github.com/hashicorp/terraform-provider-aws/blob/9bccc304967b918bf2b8873b8625f1d6973458c1/internal/service/cloudwatch/metric_alarm.go#L170-L178, has valid values of `10`, `30, and multiples of `60`. The `period` of a `metric_query`, on the other hand, can be `1`, `5`, `10`, `30`, and multiples of `60` (https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDataQuery.html)

**Pull request merged**